src/main/java/teammates/ui/webapi/QueryLogsAction.java

```diff
@@ -0,0 +1,90 @@
+package teammates.ui.webapi;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.http.HttpStatus;
+
+import com.google.logging.type.LogSeverity;
+
+import teammates.common.datatransfer.QueryLogsResults;
+import teammates.common.exception.InvalidHttpParameterException;
+import teammates.common.exception.LogServiceException;
+import teammates.common.util.Const;
+import teammates.ui.output.GeneralLogsData;
+
+/**
+ * Gets the list of error logs for a user-specified period of time.
+ */
+public class QueryLogsAction extends AdminOnlyAction {
+    private static final String DEFAULT_SEVERITIES = "INFO";
+    private static final int DEFAULT_PAGE_SIZE = 20;
+
+    private static final long TWENTY_FOUR_HOURS_IN_MILLIS = 1000L * 60 * 60 * 24;
+
+    private static final List<String> LOG_SEVERITIES = Arrays.stream(LogSeverity.values())
+            .map(Enum::toString)
+            .collect(Collectors.toList());
+
+    @Override
+    ActionResult execute() {
+        String severitiesStr;
+
+        severitiesStr = getRequestParamValue(Const.ParamsNames.QUERY_LOGS_SEVERITIES);
+        if (severitiesStr == null) {
+            severitiesStr = DEFAULT_SEVERITIES;
+        }
+
+        Instant endTime;
+        try {
+            String endTimeStr = getRequestParamValue(Const.ParamsNames.QUERY_LOGS_ENDTIME);
+            if (endTimeStr == null) {
+                endTime = Instant.now();
+            } else {
+                endTime = Instant.ofEpochMilli(Long.parseLong(endTimeStr));
+            }
```

Comment:
Can shorten slightly:
```java
        Instant endTime = Instant.now();
        try {
            String endTimeStr = getRequestParamValue(Const.ParamsNames.QUERY_LOGS_ENDTIME);
            if (endTimeStr != null) {
                endTime = Instant.ofEpochMilli(Long.parseLong(endTimeStr));
            }
```
Similarly for `startTime` later.

Reply:
OK. It is updated.